### PR TITLE
Make wildcard lookup fall back to searching everywhere in the hierarchy

### PR DESCRIPTION
**Pull request by akx**
Workaround for https://github.com/adieyal/sd-dynamic-prompts/issues/377.

This was an easier fix than actually doing search paths (since e.g. `wildcard_to_variant` makes that really messy).

### CODE REVIEW

**Comment by akx** [comment]
@adieyal Do you have time to check this out? :)

**Comment by adieyal** [comment]
> @adieyal Do you have time to check this out? :)

Sorry - i've been travelling, will take a look over the weekend

**Comment by akx** [comment]
@adieyal No worries! Hope your travels were good :)

File: src/dynamicprompts/samplers/cycle.py [hunk 1]
```diff
@@ -93,19 +96,16 @@ def _get_variant(
...
-    ):
-        values = sampling_context.wildcard_manager.get_all_values(command.wildcard)
-        new_context = sampling_context.with_variables(
+        context: SamplingContext,
+    ) -> StringGen:
+        values = context.wildcard_manager.get_all_values(command.wildcard)
+        new_context = context.with_variables(
             command.variables,
         ).with_sampling_method(SamplingMethod.CYCLICAL)
+        if len(values) == 0:
```
Comment (adieyal) [inline src/dynamicprompts/samplers/cycle.py:105]: might want to make this consistent with combinatorial.py, i.e. `if not values:`

File: src/dynamicprompts/samplers/random.py [hunk 2]
```diff
@@ -107,11 +110,9 @@ def _get_wildcard(
         values = context.wildcard_manager.get_all_values(command.wildcard)
 
         if len(values) == 0:
-            logger.warning(f"No values found for wildcard {command.wildcard}")
+            yield from get_wildcard_not_found_fallback(command, context)
```
Comment (adieyal) [inline src/dynamicprompts/samplers/random.py:113]: might want to make this consistent with combinatorial.py, i.e. if not values:

**Review by adieyal — APPROVED** [review]

**Pull request merged**